go/vt/vtgate/executor.go

```diff
@@ -368,7 +364,7 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 }
 
 func (e *Executor) destinationExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, dest key.Destination, destKeyspace string, destTabletType topodatapb.TabletType, logStats *LogStats) (*sqltypes.Result, error) {
-	return e.resolver.Execute(ctx, sql, bindVars, destKeyspace, destTabletType, dest, safeSession.Session, false /* notInTransaction */, safeSession.Options, logStats)
+	return e.resolver.Execute(ctx, sql, bindVars, destKeyspace, destTabletType, dest, safeSession, false /* notInTransaction */, safeSession.Options, logStats, true /* canAutocommit */)
```

Comment:
`destinationExec` is being used by other functions like `handleDDL`, etc. It may be better to inline this invocation directly in `handleExec`.

I was also surprised this didn't cause test failures; This is because we should increment a different (new) variable here https://github.com/vitessio/vitess/blob/master/go/vt/vttablet/sandboxconn/sandboxconn.go#L144, instead of reusing `ExecCount`.

Reply:
Oh good catch. I will inline in here.